src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -613,6 +615,90 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### Defining your own hooks for other integrations
+
+You can extend the hooks defined for Astro integration by provide the following type augmentation along with your integration code:
+
+```ts
+// your-integration/types.ts
+declare global {
+  namespace Astro {
+    interface IntegrationHooks {
+      'myLib:eventHappened': (your: string, parameters: number) => Promise<void>;
+    }
+  }
+}
+```
+
+#### Calling custom hooks on other integrations
+
+Custom hooks implemented by all the integrations in use can be accessed from the `astro:config:setup` and `astro:config:done` hooks.
+
+```ts
+// your-integration/index.ts
+import './types.ts';
+
+export default (): AstroIntegration => {
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:setup': async ({ config }) => {
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+Custom hooks don't need to be triggered inside of those hooks, they can be stored and called later:
+
+```ts
+// your-integration/index.ts
+import './types.ts';
+
+export default (): AstroIntegration => {
+  let integrations: AstroIntegration[] = [];
+
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:done': async ({ config }) => {
+        // Retrieve the integration from one of the hooks they are given
+        integrations = config.integrations;
+      },
+      'astro:build:done': async () => {
+        // Calls the custom hook at a different phase of the integration
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+#### Using a custom hook from another integration
+
+Integrations can declare hooks defined by other integrations by importing their types:
```

Comment:
This is a bit more verbose, but hopefully also slightly more precise (i.e. hooks aren’t declared _by_ importing the types, instead importing the types enables type checking etc. when the hooks are declared)

```suggestion
An integration that uses your custom hook will need to import your types file. It can then provide a value for your hook in `hooks` with type checking and editor hints.
```

Reply:
I'm a bit on the fence regarding whether this section should be directed to the author making the providing or consuming integration.

- Authors of the providing integration might refer to this when writing their own docs.
- Authors of the consuming integration might look for this information directly.

Reply:
I think in this case it’s on the integration that provides the custom hook to document using it, i.e. `fryunis-event-bus` calls a custom `event:happened` hook on any integrations defining it. It’s then on `fryunis-event-bus` to document a) that it expects `event:happened` and b) what the correct type import is. So I think these docs are educating integration authors about this pattern rather than how to use an integration that implements this pattern.